test/models/class_test.exs

```diff
@@ -1,14 +1,23 @@
 defmodule CoursePlanner.ClassTest do
   use CoursePlanner.ModelCase
 
-  alias CoursePlanner.Class
-  import CoursePlanner.Factory
+  alias CoursePlanner.{Class, Course, OfferedCourse, Repo, Terms}
 
-  @valid_attrs %{offered_course_id: nil, date: %{day: 17, month: 4, year: 2010}, finishes_at: %{hour: 14, min: 0, sec: 0}, starting_at: %{hour: 14, min: 0, sec: 0}}
+  @term_attrs %{name: "Term", start_date: "2010-01-01", end_date: "2010-12-31", status: "Active"}
+  @valid_course_attrs %{description: "some content", name: "some content", session_duration: %{hour: 14, min: 0, sec: 0}, status: "Planned"}
+  @valid_attrs %{offered_course_id: nil, date: %{day: 17, month: 4, year: 2010}, finishes_at: %{hour: 15, min: 0, sec: 0}, starting_at: %{hour: 14, min: 0, sec: 0}}
   @invalid_attrs %{}
+  @class_before_term %{offered_course_id: nil, date: %{day: 17, month: 4, year: 2009}, finishes_at: %{hour: 14, min: 0, sec: 0}, starting_at: %{hour: 14, min: 0, sec: 0}}
+  @class_after_term %{offered_course_id: nil, date: %{day: 17, month: 4, year: 2011}, finishes_at: %{hour: 14, min: 0, sec: 0}, starting_at: %{hour: 14, min: 0, sec: 0}}
+
+  defp create_course do
```

Comment:
could you use the factory `insert(:offered_course)` instead of adding this function back?